src/charm_types.py

```diff
@@ -20,3 +20,21 @@ class ExecResult(typing.NamedTuple):
     exit_code: int
     stdout: str
     stderr: str
+
+
+class DatasourcePostgreSQL(typing.TypedDict):
```

Comment:
There's only one addtional class here that doesn't seem to be used anymore. If that's the case it might make sense to remove it an rename this file to datasource.py or something similar